### PR TITLE
Fixes #15459: add a rudder agent command to force/unforce audit mode

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/15459

### CODE REVIEW

File: share/commands/agent-set-force-audit [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+# @description forbid rudder-agent to be run in non audit mode
+# @man This is useful when you want to ensure that the agent check compliance only,
+# @man and won't be doing any modification to your system. If agent is run in non 
+# @man audit, it will be automatically stopped
```
Comment (amousset) [inline share/commands/agent-set-force-audit]: Missing . at the end

File: share/commands/agent-set-force-audit [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      ;;
+    c)
+      clear_colors
+      ;;
+  esac
+done
+
+touch /opt/rudder/etc/force-audit-agent
+
+[ $? -ne 0 ] && printf "${RED}error${NORMAL}: Rudder agent could not be be forced in compliance only.\n" && exit 1
```
Comment (amousset) [inline share/commands/agent-set-force-audit]: Compliance only?

File: share/commands/agent-unset-force-audit [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      ;;
+    c)
+      clear_colors
+      ;;
+  esac
+done
+
+rm -f /opt/rudder/etc/force-audit-agent
+
+[ $? -ne 0 ] && printf "${RED}error${NORMAL}: Rudder force audit mode flag could not be removed.\n" && exit 1
```
Comment (amousset) [inline share/commands/agent-unset-force-audit]: The user does not have to know it is implemented using a flag file. We could only say something like "Rudder force audit mode could not be disabled"

**Comment by ncharles** [comment]
Commit modified

**Pull request merged**